simple-acquaintances/data/datagen.py

```diff
@@ -0,0 +1,269 @@
+        self.globalThings = np.arange(noOfGlobal)
+        self.localThings = np.arange(noOfLocal)
+
+        self.placesAffectingLocalThings = {}
+        self.__generatePALT()  # fill in placesAffectingLocalThings
+
+        self.placesLivedMean = placesLivedMean
+        self.placesLivedSD = placesLivedSD
+        self.placesLivedUpper = noOfPlaces
+        self.localLikesVariance = localLikesVariance
+        # self.truncnormGenerator = get_truncated_normal(mean, sd, 1, upper)
+
+    def __generatePALT(self):
+        """Generate porbabilities of places affecting the likable-ity of local things."""
+        # eg: [Place1:[LThing1, Lthing2], Place2:[LThing1,LThing2]]
+
+        noOfLocalThings = self.localThings.size
+        for i in self.places:
+            self.placesAffectingLocalThings[i] = np.random.uniform(
+                0, 1, noOfLocalThings)
+
+    def __getLikability(self, person):
+        """Return likeability for a localThing based on the places that the person has lived in."""
+        likeability = {}
+        for thing in self.localThings:
+            likeability[thing] = 0
+            for place in person.lived[0]:
+                truncnorm_gen = get_truncated_normal(
+                    self.placesAffectingLocalThings[place][thing], self.localLikesVariance, 0, 1)
+                likeability[thing] += truncnorm_gen.rvs()
+            likeability[thing] /= person.lived.size
+        return likeability
+
+    def __flipCoin(self, bias=0.5):
+        """Return 1 or 0 based on a biased coin toss. Default bias = 0.5."""
+        return 1 if np.random.random() < bias else 0
+
+    def __calculateSimilarity(self, p1, p2):
+        """Return similarity measure between two persons (cosine similarity [0-1])."""
+        person1 = p1.lived
+        person1 = np.concatenate(
+            (person1, np.array(list(p1.globalLikes.values()))), axis=None)
+        person1 = np.concatenate(
+            (person1, np.array(list(p1.localLikes.values()))), axis=None)
+        person1 = person1.reshape((1, -1))
+
+        person2 = p2.lived
+        person2 = np.concatenate(
+            (person2, np.array(list(p2.globalLikes.values()))), axis=None)
+        person2 = np.concatenate(
+            (person2, np.array(list(p2.localLikes.values()))), axis=None)
+        person2 = person2.reshape((1, -1))
+
+        return cosine_similarity(person1, person2)
+
+    def __generatePeople(self):
+        """Generate list of people with their feature matrices filled in."""
+        people = []
+        for i in range(self.noOfPeople):
+            person = Person()
+
+            person.index = i
+            truncnormGenerator = get_truncated_normal(
+                self.placesLivedMean, self.placesLivedSD, 1, self.placesLivedUpper)
+            noOfPlacesLived = int(truncnormGenerator.rvs())
+
+            person.lived = np.zeros(self.places.size).reshape(1, -1)
+            person.lived[0][np.random.choice(
```

Comment:
This relies on the assumption that a place's id matches its index into self.places.
Either let go of this assumption or put a comment here describing it.

(Probably same for lived.)